backend/danswer/connectors/web/connector.py

```diff
@@ -133,9 +135,20 @@ def load_from_state(self) -> GenerateDocumentsOutput:
                 title = None
                 if title_tag and title_tag.text:
                     title = title_tag.text
+                    title_tag.extract()
 
-                # Heuristics based cleaning
-                for undesired_div in ["sidebar", "header", "footer"]:
+                # Heuristics based cleaning of links
+                for undesired_links in ["muted-link"]:
+                    [
+                        tag.extract()
+                        for tag in soup.find_all(
+                            "a", class_=lambda x: x and undesired_links in x.split()
+                        )
+                    ]
+
+                # Heuristics based cleaning of divs
+                for undesired_div in ["sidebar", "header", "footer", "toctree-wrapper", "sidebar-container",
+                                      "visually-hidden", "toc-sticky", "toc-tree"]:
```

Comment:
out of curiosity, what are these? Toc-sticky, toc-tree etc? I'm not familiar with these. Are they specifically for sphinx? Please add a comment explaining what these are

Reply:
Toc is table of content. Otherwise you will repeat those in every second page and it will distract semantic search and your LLM as you will now have the same title on multiple pages. Read the docs uses sphinx (with a custom theme): https://pyserial.readthedocs.io/en/latest/index.html. On that page you can see a toctree-wrapper in the center. However, I just noticed that they use different (but similar) tags for the sidebar stuff.

I just thought about this. Maybe it would be better to add a config field next to the URL which lets you specify a bunch of XPath expressions to remove. That would be much more flexible and would support all the crazy stuff people come up with.

Reply:
I see, if you want to hardcode them here it's fine, I would just ask that you add clear comments as to what they are. The web connector is a pretty common source of headache as web scraping is messy in nature and we may need to revisit this connector again.

If you opt to have a frontend config for parsing options, please hide that under an "advanced options" type expandable so it doesn't confuse the majority of users. Similar to how you proposed supporting Auth for web connector.